src/appleseed/renderer/kernel/rendering/generic/genericframerenderer.cpp

```diff
@@ -349,13 +353,27 @@ namespace
                     }
                 }
 
-                // Run the denoiser if needed.
-                if (!m_abort_switch.is_aborted())
+                // Check abort flag.
+                if (m_abort_switch.is_aborted())
                 {
-                    m_frame.denoise(
-                        m_thread_count,
-                        m_tile_callbacks.empty() ? nullptr : m_tile_callbacks[0],
-                        &m_abort_switch);
+                    m_is_rendering = false;
+                    return;
+                }
+
+                // Denoising pass.
+                if (m_frame.get_denoising_mode() == Frame::DenoisingMode::Denoise)
+                {
+                    if (m_pass_count > 1)
```

Comment:
Why do we need this test?

Reply:
It's just cosmetic: we don't want to print these messages if there is only one pass.